File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task58.java

public class Task58 {
    public static boolean XO(String str) {
        if (str == null) {
            return true;
        }
        
        String lowerStr = str.toLowerCase();
        int countX = 0;
        int countO = 0;
        
        for (int i = 0; i < lowerStr.length(); i++) {
            char ch = lowerStr.charAt(i);
            if (ch == 'x') {
                countX++;
            } else if (ch == 'o') {
                countO++;
            }
        }
        
        return countX == countO;
    }
    
    public static void main(String[] args) {
        System.out.println("Test 1: XO(\\"ooxx\\") = " + XO("ooxx") + " (expected: true)");
        System.out.println("Test 2: XO(\\"xooxx\\") = " + XO("xooxx") + " (expected: false)");
        System.out.println("Test 3: XO(\\"ooxXm\\") = " + XO("ooxXm") + " (expected: true)");
        System.out.println("Test 4: XO(\\"zpzpzpp\\") = " + XO("zpzpzpp") + " (expected: true)");
        System.out.println("Test 5: XO(\\"zzoo\\") = " + XO("zzoo") + " (expected: false)");
    }
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task58.py

def XO(s):
    if s is None:
        return True
    
    s_lower = s.lower()
    count_x = 0
    count_o = 0
    
    for ch in s_lower:
        if ch == 'x':
            count_x += 1
        elif ch == 'o':
            count_o += 1
    
    return count_x == count_o


if __name__ == "__main__":
    print(f"Test 1: XO('ooxx') = {XO('ooxx')} (expected: True)")
    print(f"Test 2: XO('xooxx') = {XO('xooxx')} (expected: False)")
    print(f"Test 3: XO('ooxXm') = {XO('ooxXm')} (expected: True)")
    print(f"Test 4: XO('zpzpzpp') = {XO('zpzpzpp')} (expected: True)")
    print(f"Test 5: XO('zzoo') = {XO('zzoo')} (expected: False)")
